docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+  
+** TO DO: ** Resolve discrepancy between ModelInfo.location and
+ModelConfig.path.
+
+### Emitting model loading events
+
+When the `context` argument is passed to `get_model()`, it will
+retrieve the invocation event bus from the passed `InvocationContext`
+object to emit events on the invocation bus. The two events are
+"model_load_started" and "model_load_completed". Both carry the
+following payload:
+
+```
+payload=dict(
+	queue_id=queue_id,
+	queue_item_id=queue_item_id,
+	queue_batch_id=queue_batch_id,
+	graph_execution_state_id=graph_execution_state_id,
+	model_key=model_key,
+	submodel=submodel,
+	hash=model_info.hash,
+	location=str(model_info.location),
+	precision=str(model_info.precision),
+)
+```
+
+***
+
+## Get on line: The Download Queue
+
+InvokeAI can download arbitrary files using a multithreaded background
+download queue. Internally, the download queue is used for installing
+models located at remote locations. The queue is implemented by the
+`DownloadQueueService` defined in
+`invokeai.app.services.download_manager`. However, most of the
+implementation is spread out among several files in
+`invokeai/backend/model_manager/download/*`
+
+A default download queue is located in
+`ApiDependencies.invoker.services.download_queue`. However, you can
+create additional instances if you need to isolate your queue from the
+main one.
+
+### A job for every task
+
+The queue operates on a series of download job objects. These objects
+specify the source and destination of the download, and keep track of
+the progress of the download. Jobs come in a variety of shapes and
+colors as they are progressively specialized for particular download
+task.
+
+The basic job is the `DownloadJobBase`, a pydantic object with the
+following fields:
+
+| **Field**      | **Type**        |  **Default**  | **Description** |
+|----------------|-----------------|---------------|-----------------|
+| `id`             | int             |               | Job ID, an integer >= 0           |
+| `priority`       | int             | 10            | Job priority. Lower priorities run before higher priorities |
+| `source`         | Any             |               | Where to download from - untyped in base class |
+| `destination`    | Path            |               | Where to download to              |
+| `status`         | DownloadJobStatus| Idle         | Job's status (see below)          |
+| `event_handlers` | List[DownloadEventHandler]|     | Event handlers (see below)        |
+| `job_started`    | float            |              | Timestamp for when the job started running |
+| `job_ended`      | float            |              | Timestamp for when the job completed or errored out |
+| `job_sequence`   | int              |              | A counter that is incremented each time a model is dequeued |
+| `preserve_partial_downloads`| bool  | False        | Resume partial downloads when relaunched.   |
```

Comment:
`preserve_partial_downloads`? It actually isn't correctly implemented right now anyway, because the partial downloads are in a temporary directory that gets cleaned up when the app goes down. See the earlier comment thread about cleaning up partial downloads.

It should either be removed, or a robust resume system be implemented. I think I'll remove it for now.